### PR TITLE
InteractableDialog Bugfixes

### LINKED ISSUE #59: Spin sound keeps playing when reading signs
**Describe the bug**
Interacting with a sign while spinning does not stop the spin sound.

**To Reproduce**
1. Go to any room with a sign (t1r1 or t1r2 recommended)
2. Spin
3. With spin still held down, interact with a sign

**Expected behavior**
Spin sound never plays when the player is not spinning.

**Screenshots**
None, audio bug

**System Information**
 - OS/Browser: Windows 10
 - Game version: 0.2.0 alpha
 - Other specs:

### LINKED ISSUE #105: Ground pounding too close to the ground does not play landing animation
**Describe the bug**
Ground pounding too close to the ground does not play the landing animation where the screen like shakes and the stars come out of Mario

**To Reproduce**
Step 1: Hold the down arrow, Z and the arrow key opposite to your facing direction
Step 2: Press up

**Expected behavior**
Well the animation should be playing

**Screenshots**
Not necessary

**System Information**
 - OS/Browser: The one everyone uses (Windows 10)
 - Game version: v0.1.6.RC1
 - Other specs: Screen

**Additional context**
If you stand on a spot where the ground behind Mario is lower than where he is standing it won't work as you'll end up too high above the ground for the bug to happen once you get the little backward boost from backflipping

### LINKED ISSUE #115: Using FLUDD and a sign simultaneously will produce a buzzing noise
**Describe the bug**
FLUUD Groundpounding and interacting with a sign creates a loud buzzing noise, similar to that of a chainsaw, while your FLUUD shoots out tons of water.

**To Reproduce**
Press Z and Space simultaneously while in front of a sign.

**Expected behavior**
Ordinarily, you would enter the sign's text box, and your FLUUD would stop spraying out water. There shouldn't be a buzzing noise either.

**Screenshots**
image

**System Information**
 - OS/Browser: Ubuntu Linux
 - Game version: 0.20.0
 - Other specs: Not applicable

### PR DESCRIPTION
**Pull request by Kuma-Boo**
# Description of changes

- Fixed spinning and fludd when interacting with signs and toads
- Moves sign/toad's state check to the InteractableDialog class for more consistency in dialog activation

# Issue(s)

Closes #115 
Closes #59 
~~#105~~ (Will be moved to a separate pull request)

### CODE REVIEW

**Review by Koopa1018 — CHANGES_REQUESTED** [review]
Not certain about mixing multiple systems' changes like this (pound and Interactable fixes), but there's no denying they're solid fixes. We can probably get these merged pretty soon.

(Might want to look at the style guide in the meantime. `CONTRIBUTING.md`, at the root of the repo.)

**Review by Koopa1018 — CHANGES_REQUESTED** [review]
Nice work. Before we merge, would you mind making some style changes to function spacing and comment formatting? (Functions should have two lines between, and comments should have a space after the `#`.)

File: classes/interactable/sign/sign.gd [hunk 1]
```diff
@@ -15,11 +15,6 @@ func _process(delta):
 	pulse += 0.1 * delta * 60
 	material.set_shader_param("outline_color", Color(1, 1, 1, (sin(pulse) * 0.25 + 0.5) * glow_factor))
 
-
-func _state_check(body) -> bool:
```
Comment (Koopa1018) [inline classes/interactable/sign/sign.gd:19]: Tricky to see, but a line's been deleted here too

**Review by Koopa1018 — COMMENTED** [review]
Good stuff, good stuff. Keep going, is all I'd say. Also, check the PR's "Files changed" tab. Should help you find unformatted spots.

**Comment by Kuma-Boo** [comment]
Good grief, all these formatting mistakes are going to be the end of me 🤣

**Review by Koopa1018 — APPROVED** [review]
Looks good.

**Pull request merged**